chaoscenter/authentication/pkg/entities/project.go

```diff
@@ -71,6 +75,50 @@ type ListInvitationResponse struct {
 	InvitationRole MemberRole `json:"invitationRole"`
 }
 
+type ListProjectResponse struct {
+	Projects              []*Project `json:"projects"`
+	TotalNumberOfProjects *int64     `json:"totalNumberOfProjects"`
+}
+
+type Pagination struct {
+	Page  int `json:"page"`
+	Limit int `json:"limit"`
+}
+
+type ListProjectRequest struct {
+	UserID     string                  `json:"userID"`
+	Pagination *Pagination             `json:"pagination,omitempty"`
+	Sort       *SortInput              `json:"sort,omitempty"`
+	Filter     *ListProjectInputFilter `json:"filter,omitempty"`
+}
+
+type ListProjectInputFilter struct {
+	CreatedByMe     *bool   `json:"createdByMe"`
+	InvitedByOthers *bool   `json:"invitedByOthers"`
+	ProjectName     *string `json:"projectName"`
+}
+
+type ProjectSortingField string
+
+const (
+	ProjectSortingFieldName ProjectSortingField = "NAME"
+	ProjectSortingFieldTime ProjectSortingField = "TIME"
+)
+
+type SortInput struct {
+	Field     *ProjectSortingField `json:"field"`
+	Ascending *bool                `json:"ascending"`
+}
+
+const (
```

Comment:
if this is being used by the APIs, do not keep it here. create a common file where other resources(user) can also used these fields

Reply:
:) can u suggest where should I keep them